web/api/v2/create-action/index.ts

```diff
@@ -0,0 +1,119 @@
+import { errorResponse } from "@/api/helpers/errors";
+import { getAPIServiceGraphqlClient } from "@/api/helpers/graphql";
+import { verifyHashedSecret } from "@/api/helpers/utils";
+import { validateRequestSchema } from "@/api/helpers/validate-request-schema";
+import { generateExternalNullifier } from "@/lib/hashing";
+import { NextRequest, NextResponse } from "next/server";
+import * as yup from "yup";
+import { getSdk as createDynamicActionSdk } from "./graphql/create-dynamic-action.generated";
+import { getSdk as fetchApiKeySdk } from "./graphql/fetch-api-key.generated";
+
+const createActionBodySchema = yup.object({
+  app_id: yup.string().strict().required(),
+  action: yup.string().strict().required(),
+});
+
+export const POST = async (req: NextRequest) => {
+  const body = await req.json();
+  const api_key = req.headers.get("authorization")?.split(" ")[1];
+
+  if (!api_key) {
+    return errorResponse({
+      statusCode: 401,
+      code: "unauthorized",
+      detail: "API key is required.",
+      attribute: "api_key",
+      req,
+    });
+  }
+
+  const { isValid, parsedParams, handleError } = await validateRequestSchema({
+    schema: createActionBodySchema,
+    value: body,
+  });
+
+  if (!isValid) {
+    return handleError(req);
+  }
+
+  const { app_id } = parsedParams;
+  const base64ApiKey = Buffer.from(api_key, "base64").toString("utf-8");
+  const [id, secret] = base64ApiKey.split(":");
+  const client = await getAPIServiceGraphqlClient();
```

Comment:
Since we are using api key we need to be generating an API JWT so we use those permissions